modules/thermalsensor_generic.c

```diff
@@ -0,0 +1,1451 @@
+
+static void                       thermal_sensor_generic_set_depends_on     (thermal_sensor_generic_t *self, const char *sensor_name);
+static void                       thermal_sensor_generic_set_temp_offs      (thermal_sensor_generic_t *self, int offs);
+
+/* ========================================================================= *
+ * HOOKS_FOR_THERMAL_OBJECT
+ * ========================================================================= */
+
+static thermal_sensor_generic_t  *thermal_sensor_generic_from_object        (const thermal_object_t *object);
+static void                       thermal_sensor_generic_delete_cb          (thermal_object_t *object);
+static const char                *thermal_sensor_generic_get_name_cb        (const thermal_object_t *object);
+static const char                *thermal_sensor_generic_get_depends_on_cb     (const thermal_object_t *object);
+static bool                       thermal_sensor_generic_get_status_cb      (const thermal_object_t *object, THERMAL_STATUS *status, int *temp);
+static bool                       thermal_sensor_generic_get_poll_delay_cb  (const thermal_object_t *object, int *minwait, int *maxwait);
+static bool                       thermal_sensor_generic_read_sensor_cb     (thermal_object_t *object);
+
+/* ========================================================================= *
+ * CONFIGURATION_FILE
+ * ========================================================================= */
+
+static thermal_object_t          *tsg_objects_get_object          (GSList *list, const char *name);
+static thermal_object_t          *tsg_objects_add_object          (GSList **list, const char *name);
+static thermal_sensor_generic_t  *tsg_objects_add_sensor          (GSList **list, const char *name);
+static THERMAL_STATUS             tsg_objects_parse_level         (const char *key);
+static void                       tsg_objects_register_all        (GSList **list);
+static void                       tsg_objects_read_config         (GSList **list, const char *config);
+static void                       tsg_objects_quit                (GSList **list);
+static void                       tsg_objects_init                (GSList **list);
+
+/* ========================================================================= *
+ * UTILITY_FUNCTIONS
+ * ========================================================================= */
+
+/** Extract token from string buffer
+ *
+ * Leading white space at parse position is always skipped.
+ *
+ * Once parse position is at end of string, empty tokens
+ * are returned.
+ *
+ * @param pos   parse start position
+ * @param ppos  where to store position after parsing
+ * @param sep   token separator character, or -1 for any whitespace
+ *
+ * @return extracted token
+ */
+static char *
+tsg_util_slice_token(char *pos, char **ppos, int sep)
```

Comment:
Something like: strtok() the worst function in the standard c library (perhaps excluding the mbs* stuff and other fubar locale related functionality). It is so bad that I consider strtok_r() to be crap by association and pretend that neither one exists.